ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java

```diff
@@ -26,5 +26,5 @@
 /** An asynchronous output stream supporting zero buffer copying. */
 public interface DataStreamOutput extends CloseAsync<DataStreamReply> {
   /** Send out the data in the buffer asynchronously */
-  CompletableFuture<DataStreamReply> writeAsync(ByteBuffer buf);
+  CompletableFuture<DataStreamReply> writeAsync(ByteBuffer buf, boolean sync);
```

Comment:
We should add one more method
```suggestion
  default CompletableFuture<DataStreamReply> writeAsync(ByteBuffer buffer) {
    return writeAsync(buffer, false);
  }

  CompletableFuture<DataStreamReply> writeAsync(ByteBuffer buffer, boolean sync);
```